### PR TITLE
fix: Ensure (Old) Instance is removed from Launcher during reinstall

### LINKED ISSUE #859: When reinstalling an instance, it is visually duplicated
### What happened?

1. Create some instance with the "Create Pack" screen (e.g. a Vanilla 1.20.4 instance)
2. Go to "Instances"
3. Find the instance and click "Edit Instance", then "Reinstall"
4. Changing anything in the pop-up dialogue is optional
5. Click "Reinstall"

ATLauncher will duplicate the instance's entry. Both entries appear to link to the same instance, and deleting one will cause the instance's folder and the entry the deletion was requested with to disappear. Restarting the Launcher appears to fix the issue.

### Operating System

Linux

### Relevant log output

```shell
The following log output can also be found on https://paste.atlauncher.com/view/4e5805e0-4ef9-419c-a1fb-b0aa638a81dd.

[10/12/2023 11:52:40 AM] Organising filesystem
[10/12/2023 11:52:40 AM] ATLauncher Version: 3.4.35.3 [46c73c59063aff664b751c52d71b8f9d9828eadc]
[10/12/2023 11:52:40 AM] App Arguments: ["--install-method=deb"]
[10/12/2023 11:52:40 AM] JVM Arguments: ["-Dawt.useSystemAAFontSettings=on","-Dswing.aatext=true"]
[10/12/2023 11:52:40 AM] Java Version: Java 17 (17.0.9)
[10/12/2023 11:52:40 AM] Java Path: /usr/lib/jvm/java-17-openjdk-amd64
[10/12/2023 11:52:40 AM] 64 Bit Java: true
[10/12/2023 11:52:40 AM] RAM Available: 15327MB
[10/12/2023 11:52:40 AM] Launcher Directory: **USERSDIR**
[10/12/2023 11:52:40 AM] GPU: Lucienne (Advanced Micro Devices, Inc. [AMD/ATI] (0x1002)) unknown 258MB VRAM
[10/12/2023 11:52:40 AM] CPU: AMD Ryzen 5 5500U with Radeon Graphics 6 cores/12 threads
[10/12/2023 11:52:40 AM] Operating System: Debian GNU/Linux (12 (bookworm) build 6.1.0-13-amd64)
[10/12/2023 11:52:40 AM] Bitness: 64
[10/12/2023 11:52:40 AM] Uptime: 4197
[10/12/2023 11:52:40 AM] Manufacturer: GNU/Linux
[10/12/2023 11:52:40 AM] Showing splash screen and loading everything
[10/12/2023 11:52:40 AM] Checking for updated files!
[10/12/2023 11:52:41 AM] Launcher finished loading everything
[10/12/2023 11:52:41 AM] Launcher opening
[10/12/2023 11:52:41 AM] Made By Bob*
[10/12/2023 11:52:41 AM] *(Not Actually)
[10/12/2023 11:52:41 AM] Setting up Bottom Bar
[10/12/2023 11:52:41 AM] Finished Setting up Bottom Bar
[10/12/2023 11:52:41 AM] Setting up Tabs
[10/12/2023 11:52:41 AM] Finished Setting up Tabs
[10/12/2023 11:52:41 AM] Showing Launcher

COMMENT: THIS IS AFTER I DID STEP 1

[10/12/2023 11:53:04 AM] Started install of Minecraft version 1.20.4

COMMENT: THIS IS WHEN I DID STEPS 2 - 5

[10/12/2023 11:53:16 AM] Started install of Minecraft version 1.20.4
```

### Additional Information

_No response_

### Is this actually a launcher bug?

- [X] I am sure this is a bug with the launcher and not an issue with running Minecraft

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Doomsdayrs**
### Description of the Change

One line wonder.

### Testing

- [x] Reinstall instance
- [x] Reinstall instance with different version
- [x] Ensure world files preserved

### Related Issues

Closes #859

### CODE REVIEW

**Comment by Doomsdayrs** [comment]
Only issue I have is NPEs being thrown during the Install process.

File: src/main/java/com/atlauncher/workers/InstanceInstaller.java [hunk 1]
```diff
@@ -2470,6 +2470,7 @@ private void saveInstanceJson() {
 
         if (!this.isReinstall) {
             instanceLauncher = new InstanceLauncher();
+            InstanceManager.removeInstance(this.instance); // Remove old instance from app data, as it no longer exists
```
Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/workers/InstanceInstaller.java:2473]: Is this in the right place? The condition is `!this.isReinstall` so it's run only on a brand new instance being installed

Comment (Doomsdayrs) [reply src/main/java/com/atlauncher/workers/InstanceInstaller.java:2473]: Honestly I have no clue right now, I might be too sleep deprived.

Let me look at this problem again

**Comment by Doomsdayrs** [comment]
@RyanTheAllmighty I honestly have no clue why this works, but it does???

Please do check that I am not hallucinating on your own computer.

**Review by RyanTheAllmighty — APPROVED** [review]

**Pull request merged**